### PR TITLE
Fix volumetric fog in combination with spotlights

### PR DESCRIPTION
**Pull request by Geometror**
This PR fixes the artifacts which show up when using spotlights in combination with volumetric fog. 
Fixes both artifacts described in #55527.
(#41357 might be fixed as well for spotlights, needs testing)

Screenshot 2021-12-08 013220

fix-spotlights-fog

### CODE REVIEW

File: servers/rendering/renderer_rd/renderer_scene_render_rd.cpp [hunk 1]
```diff
@@ -4417,7 +4417,7 @@ void RendererSceneRenderRD::_update_volumetric_fog(RID p_render_buffers, RID p_e
 
 		uint32_t cluster_screen_width = (rb->width - 1) / cluster_size + 1;
 		uint32_t cluster_screen_height = (rb->height - 1) / cluster_size + 1;
-		params.cluster_type_size = cluster_screen_width * cluster_screen_height * (32 + 32);
+		params.cluster_type_size = cluster_screen_width * cluster_screen_height * (16 + 32);
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/renderer_scene_render_rd.cpp:4420]: I wonder if this should be the same as when doing the regular scene render i.e.: https://github.com/godotengine/godot/blob/547c27077721cf024031bedd69e55268c8d6b410/servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp#L626

**Review by clayjohn — COMMENTED** [review]
Changes to the shadow lookup look good. They are the same as what I did for omnilights in #53353 and what JFons did for scene shaders in 51335

I left a comment about the cluster_type_size I am unsure what the number should be, but my guess is matching the scene shader definition will be best

File: servers/rendering/renderer_rd/renderer_scene_render_rd.cpp (hunk 1 above)
Comment (Geometror) [reply servers/rendering/renderer_rd/renderer_scene_render_rd.cpp]: You're right, this should be (params.max_cluster_element_count_div_32 + 32). Changed and tested it accordingly, seems to work as expected.

File: servers/rendering/renderer_rd/renderer_scene_render_rd.cpp [hunk 2]
```diff
@@ -4417,9 +4417,10 @@ void RendererSceneRenderRD::_update_volumetric_fog(RID p_render_buffers, RID p_e
 
 		uint32_t cluster_screen_width = (rb->width - 1) / cluster_size + 1;
 		uint32_t cluster_screen_height = (rb->height - 1) / cluster_size + 1;
-		params.cluster_type_size = cluster_screen_width * cluster_screen_height * (32 + 32);
-		params.cluster_width = cluster_screen_width;
+
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/renderer_scene_render_rd.cpp:4420]: This whitespace can be deleted, just to keep things clean

Comment (Geometror) [reply servers/rendering/renderer_rd/renderer_scene_render_rd.cpp]: Ok, done.

**Review by clayjohn — APPROVED** [review]

**Pull request merged**